sklearn/ensemble/tests/test_stacking.py

```diff
@@ -838,3 +839,15 @@ def test_get_feature_names_out(
 
     names_out = stacker.get_feature_names_out(feature_names)
     assert_array_equal(names_out, expected_names)
+
+
+def test_stacking_classifier_base_regressor():
+    """Check that a regressor can be estimator for a classifier."""
+    X_train, X_test, y_train, y_test = train_test_split(
+        scale(X_iris), y_iris, stratify=y_iris, random_state=42
+    )
+    clf = StackingClassifier(estimators=[("ridge", Ridge())])
+    clf.fit(X_train, y_train)
```

Comment:
We should add a test where `y` is encoded with string.

For instance using 

```python
y_str = iris.feature_names[iris.target]
```

Reply:
No error is risen from the test I implemented given that StackingClassifier uses a LabelEncoder during the fit method. I understand that label encoder cannot preserve the order so I should add a check there.

My idea is trying something like this:

```python
        for est in estimators:
            if est != "drop" and not is_regressor(est):
               y = check_array(y, ensure_2d=False, dtype="numeric")
```

However the error message is not very informative, it's something like:

`ValueError: dtype='numeric' is not compatible with arrays of bytes/strings.Convert your data to numeric values explicitly instead.`

I checked the code from `check_array` and `input_name` is ignored when checking the dtype numeric.

Should I change `check_array` for a more informative error, catch the error from `check_array` in the fit or just leave it like that?